### ISSUE TITLE
Subscribe to Feed button in Safari opens Apple News

### ISSUE THREAD

**Issue by loupgaroublond** [Ranchero-Software/NetNewsWire#1435]
Basically this issue is what it says on the tin. When I click the Subscribe to Feed button, NetNewsWire opens up things in Apple News. Apple News then pops up an error saying it can't subscribe to that.

I noticed in my settings, NetNewsWire shows the default RSS reader as Apple News. When I change this setting to NetNewsWire, subscribing to feeds works.

This is a bug because:
- The app providing the extension providing the button is NetNewsWire. It's not obvious that it would open a different app instead.
- When the user encounters unexpected behavior, ie opening the wrong app, it's not obvious that the setting is in the app itself

I'm guessing this is because NetNewsWire tries to use the built in facilities to open a URL. I'm hoping the UX can be fixed, or patched over with some helpful messages in the right place, without pushing you fine folks to implementing your own IPC system.

**Comment by Wevah** [comment]
In lieu of custom IPC, the extension could just use a custom URL scheme.

**Comment by danielpunkass** [comment]
Hi @loupgaroublond - I am the person who implemented the Subscribe to Feed feature in NetNewsWire. When I added the feature, it was partly inspired by the fact that my previous Subscribe to Feed standalone Safari extension was being made unusable by Apple's addition of warning prompts every time you open a URL that will be handled by another app. Since I love NetNewsWire and I wanted to keep "my plugin" alive, I thought this was the perfect way to do it.

The reason I intentionally kept its implementation the way it is, is because I wanted to leave open the possibility that the extension could continue to serve anybody who used the old extension, even if they choose not to use NetNewsWire as their default news reader. Whatever news reader a person uses, they have the option now to simply download NetNewsWire, activate the extension, and never open NetNewsWire again, if that's what they choose.

So, I personally think it's a useful and generous approach that the plugin should behave the way it does, at least as an option.

I can see how there are some usability issues arising out of this. I didn't consider that Apple News would take the role of default RSS reader. It does seem like it will be very likely that many people don't have NetNewsWire as their default reader at least the first time they install the app.

So I guess what I'm saying is I think we should address this by serving NetNewsWire users first: it probably should default to opening in NetNewsWire regardless of the default news reader settiong, but also by serving the greater commununity by offering a toggle setting or something in Preferences:

Safari Extension:
     ( ) Opens new feeds in NetNewsWire
     ( ) Opens new feeds in system default news app

I know that most apps would not consider this edge case of providing a Safari app extension to the whole community of RSS apps, but I think the spirit of NetNewsWire, and the unique history of this plugin as I previously developed it, makes it a more obvious choice to continue that public service.

**Comment by danielpunkass** [comment]
I'm going to work on this when I can, probably by adding a NNW preference along the lines of what I suggested above. Thanks for the report @loupgaroublond !

**Comment by loupgaroublond** [comment]
This all sounds great.  I do appreciate there's a serious power user case that works for the whole community.

A couple things to also note, first, I don't remember exactly what happened a couple months ago when I first installed NNW to try it out. I may have been prompted to make it the default news reader and opted not to, since I was seeing if it would be useful to me.

Second, I can imagine any number of ways this setting could get changed on someone without their knowledge.

Basically, I got stumped by this, so I totally expect someone way less computer-savvy to have even broader issues with it. Happy to file the report!

**Comment by defjaf** [comment]
I'm not completely sure this belongs here, but is there currently a way to set the default RSS reader app? [This page](https://netnewswire.com/help/mac/5.0/en/safari-extension) implies that we can change it in the prefs, but it doesn't appear to be the case in 6.0.2.

**Comment by danielpunkass** [comment]
@defjaf I'll take a look. I sort of lost track of this issue ... <looks up at clock> ... two years ago.

**Comment by brentsimmons** [comment]
Since we can’t set the default reader due to sandboxing, should the Subscribe to Feed button use a NetNewsWire-specific URL scheme instead of feed:?

**Comment by danielpunkass** [comment]
@brentsimmons Is that what happened? We lost the default news setting with sandboxing?

**Comment by brentsimmons** [comment]
Yes — sandboxing won’t let us set a default RSS reader anymore.
